Tools/plot_parallel.py

```diff
@@ -100,25 +106,27 @@ def plot_snapshot(filename):
         plt.colorbar()
         plt.xlim(ds.domain_left_edge[dim-1], ds.domain_right_edge[dim-1])
         plt.ylim(ds.domain_left_edge[0], ds.domain_right_edge[0])
-        if args.use_vmax:
-            plt.clim(-args.vmax, args.vmax)
+        if vmax is not None:
+            plt.clim(-vmax, vmax)
     if plotlib == 'yt':
         # Directly plot with yt
         sl = yt.SlicePlot(ds, yt_slicedir[dim], args.field, aspect=yt_aspect[dim])
+        if vmax is not None:
+            sl.set_zlim(-vmax, vmax)
 
     # Plot particle quantities
     for ispecies, pspecies in enumerate(pslist):
         if pspecies in [x[0] for x in ds.field_list]:
             if plotlib == 'matplotlib':
                 # Read particle quantities from yt dataset
-                xp = ad[pspecies, 'particle_position_x'].v
+                xp = all_data_level_0[pspecies, 'particle_position_x'].v
```

Comment:
I understand that there was a bug here, but I am not sure this new version would work in the presence of MR. Could we revert from `all_data_level_0` to `ad` for particle quantities, and add `ad = ds.all_data()` right after `if plotlib == 'matplotlib':` (I accidentally deleted it)?
Thanks for catching it!

Reply:
Yes, this can be changed to use ds.all_data()